lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -1218,13 +1224,14 @@ MemoryBufferSerializedModuleLoader::loadModule(SourceLoc importLoc,
     return nullptr;
 
   auto moduleID = path[0];
+  auto moduleName = Ctx.getRealModuleName(moduleID.Item).str();
```

Comment:
Is the idea that the real module name is a long unique identifier (com.test.foo) and that path[0] is a short name (foo)?

Are module imports in serializes Swift modules using the long unique name?

MemoryBufferSerializedModuleLoader is only used by LLDB and with LLDB's global view of the application it may not generally be possible to unambiguously perform this translation. But if we only need to deal imports a user types into the expression evaluator then the right solution would be to intercept them in SwiftExpressionParser and rewrite them to the long unique name there, where we still have the contextual information.

Reply:
No it's not a long unique name. The module aliasing feature here allows a scenario like this: 
If you're building `MyModule`, and it has `import Foo`, but we want to alias `Foo` with another name under the hood due to a name collision, then we want to be able to run  `swift-frontend -module-name MyModule -module-alias Foo=Bar`, where the name `Bar` is the actual name of `Foo`, and there's Bar.swiftmodule on disk. The `getRealName()` returns `Bar`, while `getName()` returns `Foo`. If there's no `-module-alias` option used, `getRealName()` and `getName()` will have the same value.

Reply:
Each translation unit already stores all imported modules in the debug info. We could represent a module alias using a DIImportedEntity("short alias", DIModule("unique name")).
When the expression evaluator sees in import statement it checks the debug info for an alias and substituted the long name in the AST, so MemoryBufferSerializedModuleLoader only sees the long unique name.

If you agree that this would work, then we don't need to modify MemoryBufferSerializedModuleLoader in this patch.

Reply:
If a module a user typed in to import is not loaded yet, it won't work as the alias hasn't been stored, in which case they'll will have to import the physical module name explicitly until after it's loaded and the alias is stored. I'm not sure if the behavior seems consistent enough or might cause confusion.